### PR TITLE
SYMPOWER-4917 CompositeLogListener improvements

### PR DESCRIPTION
**Pull request by neeme-praks-sympower**

### CODE REVIEW

File: slf4j-simple/src/main/java/org/slf4j/impl/SkipRepeatingMessagesLogStream.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  private final PrintStream logStream;
+  private final String prefix;
+
+  private String lastErrorMessage;
+  private Exception lastException;
+
+  public SkipRepeatingMessagesLogStream(
+      SimpleMicroDateFormat dateFormat,
+      SimpleTimeProvider timeProvider,
+      StackTracePrinter stackTracePrinter, PrintStream logStream,
```
Comment (tonisojandu-sympower) [inline slf4j-simple/src/main/java/org/slf4j/impl/SkipRepeatingMessagesLogStream.java]: I think new line for PrintStream here would help with readability.

File: slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            "THROW_EXCEPTION,  RETURN_FALSE,     false",
+            "THROW_EXCEPTION,  RETURN_TRUE,      false",
+            "RETURN_FALSE,     THROW_EXCEPTION,  false",
+            "RETURN_TRUE,      THROW_EXCEPTION,  false",
+    })
+    void logTwoListenersException(LogListenerRecipe givenDelegateListener1, LogListenerRecipe givenDelegateListener2, boolean expectedResult) {
+      logListener(expectedResult, givenDelegateListener1, givenDelegateListener2);
+    }
+
+    void logListener(boolean expectedResult, LogListenerRecipe... listenerRecipes) {
```
Comment (tonisojandu-sympower) [inline slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java:109]: I would change naming and order a bit here to help with readability:
```
void whenLogging(LogListenerRecipe... givenListenerRecipes, boolean thenExpectResult)
```

File: slf4j-simple/src/test/java/org/slf4j/impl/SimpleLogListenerWithExpectations.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SimpleLogListenerWithExpectations {
+
+  private final SimpleLogListener logListener = mock(SimpleLogListener.class);
+  private Exception thrownException;
+  private boolean expectIsAliveCall = false;
+
+  SimpleLogListenerWithExpectations whenLogIsCalledThenReturn(boolean isSuccessfulReturnValue) {
```
Comment (tonisojandu-sympower) [inline slf4j-simple/src/test/java/org/slf4j/impl/SimpleLogListenerWithExpectations.java]: I suggest not mixing BDDMockito with regular Mockito. I suggest changing these methods to:
```
SimpleLogListenerWithExpectations givenAListenerThatWillOnLogCommandReturn(boolean isSuccessfulReturnValue) {
   given(logListener.log(any(SimpleLogListener.SimpleLogEvent.class))).willReturn(isSuccessfulReturnValue);
}
```

**Review by tonisojandu-sympower — CHANGES_REQUESTED** [review]

File: slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java (hunk 2 above)
Comment (neeme-praks-sympower) [reply slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java:109]: Unfortunately, varargs has to be the last argument.

File: slf4j-simple/src/test/java/org/slf4j/impl/SimpleLogListenerWithExpectations.java (hunk 3 above)
Comment (neeme-praks-sympower) [reply slf4j-simple/src/test/java/org/slf4j/impl/SimpleLogListenerWithExpectations.java]: Improved

File: slf4j-simple/src/main/java/org/slf4j/impl/SkipRepeatingMessagesLogStream.java (hunk 1 above)
Comment (neeme-praks-sympower) [reply slf4j-simple/src/main/java/org/slf4j/impl/SkipRepeatingMessagesLogStream.java]: Good catch!

File: slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java (hunk 2 above)
Comment (tonisojandu-sympower) [reply slf4j-simple/src/test/java/org/slf4j/impl/CompositeLogListenerTest.java:109]: :cry:

**Review by tonisojandu-sympower — APPROVED** [review]
Looks good!

**Review by tonisojandu-sympower — APPROVED** [review]

**Pull request merged**